contracts/strategies/convex/base/ConvexStrategy2TokenHodl.sol

```diff
@@ -0,0 +1,442 @@
+  */
+  function salvage(address recipient, address token, uint256 amount) external onlyControllerOrGovernance {
+     // To make sure that governance cannot come in and take away the coins
+    require(!unsalvagableTokens(token), "token is defined as not salvagable");
+    IERC20(token).safeTransfer(recipient, amount);
+  }
+
+  /*
+  *   Get the reward, sell it in exchange for underlying, invest what you got.
+  *   It's not much, but it's honest work.
+  *
+  *   Note that although `onlyNotPausedInvesting` is not added here,
+  *   calling `investAllUnderlying()` affectively blocks the usage of `doHardWork`
+  *   when the investing is being paused by governance.
+  */
+  function doHardWork() external onlyNotPausedInvesting restricted {
+    IBaseRewardPool(rewardPool()).getReward();
+    _liquidateReward();
+    investAllUnderlying();
+  }
+
+  /**
+  * Can completely disable claiming UNI rewards and selling. Good for emergency withdraw in the
+  * simplest possible way.
+  */
+  function setSell(bool s) public onlyGovernance {
+    _setSell(s);
+  }
+
+  /**
+  * Sets the minimum amount of CRV needed to trigger a sale.
+  */
+  function setSellFloor(uint256 floor) public onlyGovernance {
+    _setSellFloor(floor);
+  }
+
+  // masterchef rewards pool ID
+  function _setPoolId(uint256 _value) internal {
+    setUint256(_POOLID_SLOT, _value);
+  }
+
+  function poolId() public view returns (uint256) {
+    return getUint256(_POOLID_SLOT);
+  }
+
+  function _setDepositToken(address _address) public onlyGovernance {
+    setAddress(_DEPOSIT_TOKEN_SLOT, _address);
+  }
+
+  function depositToken() public view returns (address) {
+    return getAddress(_DEPOSIT_TOKEN_SLOT);
+  }
+
+  function _setDepositReceipt(address _address) internal {
+    setAddress(_DEPOSIT_RECEIPT_SLOT, _address);
+  }
+
+  function depositReceipt() public view returns (address) {
+    return getAddress(_DEPOSIT_RECEIPT_SLOT);
+  }
+
+  function _setDepositArrayPosition(uint256 _value) public onlyGovernance {
+    setUint256(_DEPOSIT_ARRAY_POSITION_SLOT, _value);
+  }
+
+  function depositArrayPosition() public view returns (uint256) {
+    return getUint256(_DEPOSIT_ARRAY_POSITION_SLOT);
+  }
+
+  function _setCurveDeposit(address _address) public onlyGovernance {
+    setAddress(_CURVE_DEPOSIT_SLOT, _address);
+  }
+
+  function curveDeposit() public view returns (address) {
+    return getAddress(_CURVE_DEPOSIT_SLOT);
+  }
+
+  function _setPotPool(address _address) public onlyGovernance {
+    setAddress(_POTPOOL_SLOT, _address);
+  }
+
+  function potPool() public view returns (address) {
+    return getAddress(_POTPOOL_SLOT);
+  }
+
+  function _setHodlVault(address _address) public onlyGovernance {
+    setAddress(_HODLVAULT_SLOT, _address);
+  }
+
+  function hodlVault() public view returns (address) {
+    return getAddress(_HODLVAULT_SLOT);
+  }
+
+  function _setHodlRatio(uint256 _value) public onlyGovernance {
```

Comment:
Let's use underscores only for internal/private methods.
If a method is public, let's remove the underscore prefixes.